pennylane/workflow/qnode.py

```diff
@@ -551,7 +551,7 @@ def __init__(
         # input arguments
         self.func = func
         self.device = device
-        self._interface = get_canonical_interface_name(interface)
+        self._interface = get_canonical_interface_name(interface, _validate_jax_version=True)

```

Comment:
Wouldn't it be less invasive to do something like,
```suggestion
        self._interface = get_canonical_interface_name(interface)
        if "jax" in self.interface:
            _validate_jax_version()
```
and not have to modify anything downstream. For example, `_resolve_interface` is called in `qml.execute` which (with the new changes coming) will only be configurable purely from the `QNode` with no other entry point. That way I see us only having to verify the version here.